src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private Object[] storage = new Object[INITIAL_CAPACITY];
+    private int currentTailIndex = 0;
+
     @Override
     public void add(T value) {
-
+        growIfArrayFull();
+        storage[currentTailIndex] = value;
+        currentTailIndex++;
     }
 
     @Override
     public void add(T value, int index) {
+        growIfArrayFull();
+
+        if (index < 0 || index > currentTailIndex) {
+            throw new ArrayListIndexOutOfBoundsException("Can't add by index " + index
+                + "while current index is " + currentTailIndex);
+        }
 
```

Comment:
1. It's better to grow array after index check
2. It's better to reuse existing methods
```suggestion
        if (index == currentTailIndex) {
            add(value);
            return;
        }
        checkIfIndexExists(index);
        growIfArrayFull();
```

Reply:
agree, i just have stuck with set while could just use add... )))